# Compet - Formação 4.0

Aluna: Ruth Xavier

### Carregando o conjunto de dados

**Base escolhida:** Adequação da Formação Docente (2024)

**Descrição:** Percentual de docentes na Educação Infantil (Creche e Pré-Escola), no Ensino Fundamental, no Ensino Médio e Educação de Jovens e Adultos por grupo de adequação da formação à disciplina que leciona, segundo localização e dependência administrativa - Brasil, Regiões Geográficas e Unidades da Federação - 2024

#### Dicionário dos dados: 

01. NU_ANO_CENSO: Ano em que foi realizado o censo;
02. UNIDGEO: Unidade Geográfica; 
03. NO_CATEGORIA: Categorizando em zona Rural, Urbana e Total; 
04. NO_DEPENDENCIA: Dependência administrativa; 
05. ED_INF_CAT_1: Educação Infantil Grupo 1; 
06. ED_INF_CAT_2: Educação Infantil Grupo 2; 
07. ED_INF_CAT_3: Educação Infantil Grupo 3; 
08. ED_INF_CAT_4: Educação Infantil Grupo 5;
09. FUN_CAT_1: Ensino Fundamental Grupo 1 total;  
10. FUN_CAT_2: Ensino Fundamental Grupo 2 total;
11. FUN_CAT_3: Ensino Fundamental Grupo 3 total; 
12. FUN_CAT_4: Ensino Fundamental Grupo 4 total;  
13. FUN_CAT_5: Ensino Fundamental Grupo 5 total; 
14. FUN_AI_CAT_1: Ensino Fundamental Grupo 1 - Anos Iniciais; 
15. FUN_AI_CAT_2: Ensino Fundamental Grupo 2 - Anos Iniciais; 
16. FUN_AI_CAT_3: Ensino Fundamental Grupo 3 - Anos Iniciais; 
17. FUN_AI_CAT_4: Ensino Fundamental Grupo 4 - Anos Iniciais; 
18. FUN_AI_CAT_5: Ensino Fundamental Grupo 5 - Anos Iniciais; 
19. FUN_AF_CAT_1: Ensino Fundamental Grupo 1 - Anos Finais;
20. FUN_AF_CAT_2: Ensino Fundamental Grupo 2 - Anos Finais;
21. FUN_AF_CAT_3: Ensino Fundamental Grupo 3 - Anos Finais;
22. FUN_AF_CAT_4: Ensino Fundamental Grupo 4 - Anos Finais;
23. FUN_AF_CAT_5: Ensino Fundamental Grupo 5 - Anos Finais;
24. MED_CAT_1: Ensino Médio Grupo 1; 
25. MED_CAT_2: Ensino Médio Grupo 2;
26. MED_CAT_3: Ensino Médio Grupo 3;
27. MED_CAT_4: Ensino Médio Grupo 4;
28. MED_CAT_5: Ensino Médio Grupo 5;
29. EJA_FUN_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Fundamental;
30. EJA_FUN_CAT_2: Educação de Jovens e Adultos Grupo 2 - Ensino Fundamental;
31. EJA_FUN_CAT_3: Educação de Jovens e Adultos Grupo 3 - Ensino Fundamental;
32. EJA_FUN_CAT_4: Educação de Jovens e Adultos Grupo 4 - Ensino Fundamental;
33. EJA_FUN_CAT_5: Educação de Jovens e Adultos Grupo 5 - Ensino Fundamental;
34. EJA_MED_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Médio;
35. EJA_MED_CAT_2: Educação de Jovens e Adultos Grupo 2 - Ensino Médio;
36. EJA_MED_CAT_3: Educação de Jovens e Adultos Grupo 3 - Ensino Médio;
37. EJA_MED_CAT_4: Educação de Jovens e Adultos Grupo 4 - Ensino Médio;
38. EJA_MED_CAT_5: Educação de Jovens e Adultos Grupo 5 - Ensino Médio;
39. EJA_MED_CAT_1: Educação de Jovens e Adultos Grupo 1 - Ensino Médio;

GRUPO 1: Docentes com formação superior de licenciatura na mesma disciplina que lecionam, ou bacharelado na mesma disciplina com curso de complementação pedagógica concluído;

GRUPO 2: Docentes com formação superior de bacharelado na disciplina
correspondente, mas sem licenciatura ou complementação pedagógica;

GRUPO 3: Docentes com licenciatura em área diferente daquela que lecionam, ou com bacharelado nas disciplinas da base curricular comum e complementação pedagógica concluída em área diferente daquela que lecionam;

GRUPO 4: Docentes com outra formação superior não considerada nas categorias anteriores;

GRUPO 5: Docentes que não possuem curso superior completo.

In [34]:
# Importando bibliotecas

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 

In [35]:
afd = pd.read_excel('datasets/AFD_BRASIL_REGIOES_UFS_2024.xlsx', skiprows=10)

In [36]:
afd.head()

NU_ANO_CENSO UNIDGEO NO_CATEGORIA NO_DEPENDENCIA ED_INF_CAT_1 ED_INF_CAT_2  \
0         2024  Brasil        Total          Total         64.1          0.7   
1         2024  Brasil       Urbana          Total         64.7          0.7   
2         2024  Brasil        Rural          Total         57.9          0.9   
3         2024  Brasil        Total        Federal         56.4          2.3   
4         2024  Brasil       Urbana        Federal         55.8          2.4   

  ED_INF_CAT_3 ED_INF_CAT_4 ED_INF_CAT_5 FUN_CAT_1  ... EJA_FUN_CAT_1  \
0         17.8          1.8         15.6      68.8  ...            27   
1         18.2          1.9         14.5      73.3  ...          35.5   
2         14.1          1.7         25.4      46.6  ...            10   
3         26.3          7.7          7.3      87.2  ...          68.9   
4         26.7          7.9          7.2      87.2  ...          68.9   

  EJA_FUN_CAT_2 EJA_FUN_CAT_3 EJA_FUN_CAT_4 EJA_FUN_CAT_5 EJA_MED_CAT_1  \
0           0.5          44.6           2.4          25.5          56.1   
1           0.7          47.9           2.7          13.2          57.9   
2           0.1          38.1           1.8            50          38.4   
3             6          17.5           5.5           2.1          83.3   
4             6          17.5           5.5           2.1          83.2   

  EJA_MED_CAT_2 EJA_MED_CAT_3 EJA_MED_CAT_4 EJA_MED_CAT_5  
0           1.1          36.1           3.1           3.6  
1           1.2          35.1           3.1           2.7  
2           0.6            46           3.6          11.4  
3           6.7           6.4           2.9           0.7  
4           6.8           6.3             3           0.7  

[5 rows x 39 columns]

In [37]:
afd.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 39 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   NU_ANO_CENSO    594 non-null    object
 1   UNIDGEO         588 non-null    object
 2   NO_CATEGORIA    588 non-null    object
 3   NO_DEPENDENCIA  588 non-null    object
 4   ED_INF_CAT_1    588 non-null    object
 5   ED_INF_CAT_2    588 non-null    object
 6   ED_INF_CAT_3    588 non-null    object
 7   ED_INF_CAT_4    588 non-null    object
 8   ED_INF_CAT_5    588 non-null    object
 9   FUN_CAT_1       588 non-null    object
 10  FUN_CAT_2       588 non-null    object
 11  FUN_CAT_3       588 non-null    object
 12  FUN_CAT_4       588 non-null    object
 13  FUN_CAT_5       588 non-null    object
 14  FUN_AI_CAT_1    588 non-null    object
 15  FUN_AI_CAT_2    588 non-null    object
 16  FUN_AI_CAT_3    588 non-null    object
 17  FUN_AI_CAT_4    588 non-null    object
 18  FUN_AI_CAT

In [38]:
# Verificando a existência de dados nulos

afd.isna().any().any()

np.True_

In [39]:
#Observando o total de nulos por coluna

afd.isna().sum()

NU_ANO_CENSO      1
UNIDGEO           7
NO_CATEGORIA      7
NO_DEPENDENCIA    7
ED_INF_CAT_1      7
ED_INF_CAT_2      7
ED_INF_CAT_3      7
ED_INF_CAT_4      7
ED_INF_CAT_5      7
FUN_CAT_1         7
FUN_CAT_2         7
FUN_CAT_3         7
FUN_CAT_4         7
FUN_CAT_5         7
FUN_AI_CAT_1      7
FUN_AI_CAT_2      7
FUN_AI_CAT_3      7
FUN_AI_CAT_4      7
FUN_AI_CAT_5      7
FUN_AF_CAT_1      7
FUN_AF_CAT_2      7
FUN_AF_CAT_3      7
FUN_AF_CAT_4      7
FUN_AF_CAT_5      7
MED_CAT_1         7
MED_CAT_2         7
MED_CAT_3         7
MED_CAT_4         7
MED_CAT_5         7
EJA_FUN_CAT_1     7
EJA_FUN_CAT_2     7
EJA_FUN_CAT_3     7
EJA_FUN_CAT_4     7
EJA_FUN_CAT_5     7
EJA_MED_CAT_1     7
EJA_MED_CAT_2     7
EJA_MED_CAT_3     7
EJA_MED_CAT_4     7
EJA_MED_CAT_5     7
dtype: int64

Através da visualização dos dados com afd.head() e afd.info() e afd.isna().any().any() foi possível identificar que: 

- Há 39 colunas e 595 linhas; 
- A maioria das colunas possuem 588 valores que não são linhas, logo 6 linhas estão vazias (devido ao skiprow, geralmente exibe uma linha a mais no final); 
- E todas as colunas foram identificadas como object, pois provavelmente há valores não numéricos como o uso do "--" ou que estejam vazias.

## Preparação dos dados para análise

A base de dados referesse a uma coleta de dados realizada em 2024 a cerca do indiciador educacional a cerca da adequação da formação docente, logo isso significa que essa base tem por objetivo mapear se os docentes tem uma formação em conformidade entre a disciplina que lecionam e suas respectivas formações acadêmicas. 

De acordo com a LDB, Lei de Diretrizes e Base da Educação, Lei nº9.394/1996, no artigo 62, que contempla os dados presentes:

    Art. 62. A formação de docentes para atuar na educação básica far-se-á em nível superior, em curso de licenciatura, de graduação plena, em universidades e institutos superiores de educação, admitida, como formação mínima para o exercício do magistério na educação infantil e nos 5 (cinco) primeiros anos do ensino fundamental, a oferecida em nível médio na modalidade normal.

**Fazendo um recorte nos dados**

Resolvi definir um foco principal: Educação de Jovens e Adultos - Ensino Fundamental e Ensino Médio


### Selecionando dados  

As colunas selecionadas para a análise:

*possíveis filtros* 
UNIDGEO, NO_CATEGORIA, NO_DEPENDENCIA

*Etapa de ensino - Educação de Jovens e Adultos - Ensino Fundamental*
EJA_FUN_CAT_1, EJA_FUN_CAT_2, EJA_FUN_CAT_3, EJA_FUN_CAT_4, EJA_FUN_CAT_5

*Etapa de ensino - Educação de Jovens e Adultos - Ensino Médio*
EJA_MED_CAT_1,EJA_MED_CAT_2, EJA_MED_CAT_3, EJA_MED_CAT_4, EJA_MED_CAT_5


In [40]:
afd_eja = afd.iloc[:, [1, 2, 3, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38]]

afd_eja

UNIDGEO NO_CATEGORIA NO_DEPENDENCIA EJA_FUN_CAT_1 EJA_FUN_CAT_2  \
0    Brasil        Total          Total            27           0.5   
1    Brasil       Urbana          Total          35.5           0.7   
2    Brasil        Rural          Total            10           0.1   
3    Brasil        Total        Federal          68.9             6   
4    Brasil       Urbana        Federal          68.9             6   
..      ...          ...            ...           ...           ...   
590     NaN          NaN            NaN           NaN           NaN   
591     NaN          NaN            NaN           NaN           NaN   
592     NaN          NaN            NaN           NaN           NaN   
593     NaN          NaN            NaN           NaN           NaN   
594     NaN          NaN            NaN           NaN           NaN   

    EJA_FUN_CAT_3 EJA_FUN_CAT_4 EJA_FUN_CAT_5 EJA_MED_CAT_1 EJA_MED_CAT_2  \
0            44.6           2.4          25.5          56.1           1.1   
1            47.9           2.7          13.2          57.9           1.2   
2            38.1           1.8            50          38.4           0.6   
3            17.5           5.5           2.1          83.3           6.7   
4            17.5           5.5           2.1          83.2           6.8   
..            ...           ...           ...           ...           ...   
590           NaN           NaN           NaN           NaN           NaN   
591           NaN           NaN           NaN           NaN           NaN   
592           NaN           NaN           NaN           NaN           NaN   
593           NaN           NaN           NaN           NaN           NaN   
594           NaN           NaN           NaN           NaN           NaN   

    EJA_MED_CAT_3 EJA_MED_CAT_4 EJA_MED_CAT_5  
0            36.1           3.1           3.6  
1            35.1           3.1           2.7  
2              46           3.6          11.4  
3             6.4           2.9           0.7  
4             6.3             3           0.7  
..            ...           ...           ...  
590           NaN           NaN           NaN  
591           NaN           NaN           NaN  
592           NaN           NaN           NaN  
593           NaN           NaN           NaN  
594           NaN           NaN           NaN  

[595 rows x 13 columns]

In [41]:
# Dicionários com as colunas selecionadas para análise

data_dictionary = pd.DataFrame([
    {
        'Variável': 'GEO',
        'Descrição': 'Unidade Geográfica',
        'Tipo': 'Categórico', 
        'Subtipo': 'Nominal',
        'Obs': 'Região do Brasil'
    },
    {
        'Variável': 'LOC',
        'Descrição': 'Categoria',
        'Tipo': 'Categórico', 
        'Subtipo': 'Nominal',
        'Obs': 'Zona Rural e Urbana'
    },
    {
        'Variável': 'DA',
        'Descrição': 'Dependência Administrativa',
        'Tipo': 'Categórico', 
        'Subtipo': 'Ordinal',
        'Obs': 'Municipal, Estudal, Federal e Privado'
    },
    {
        'Variável': 'EJAFUNG1',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 1',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG2',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 2',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG3',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 3',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG4',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 4',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAFUNG5',
        'Descrição': 'EJA - Ensino Fundamental - Grupo 5',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG1',
        'Descrição': 'EJA - Ensino Médio - Grupo 1',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG2',
        'Descrição': 'EJA - Ensino Médio - Grupo 2',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG3',
        'Descrição': 'EJA - Ensino Médio - Grupo 3',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG4',
        'Descrição': 'EJA - Ensino Médio - Grupo 4',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    },
    {
        'Variável': 'EJAMEDG5',
        'Descrição': 'EJA - Ensino Médio - Grupo 5',
        'Tipo': 'Numérico', 
        'Subtipo': 'Contínuo',
        'Obs': ''
    }
])

In [42]:
data_dictionary

Variável                           Descrição        Tipo   Subtipo  \
0        GEO                  Unidade Geográfica  Categórico   Nominal   
1        LOC                           Categoria  Categórico   Nominal   
2         DA          Dependência Administrativa  Categórico   Ordinal   
3   EJAFUNG1  EJA - Ensino Fundamental - Grupo 1    Numérico  Contínuo   
4   EJAFUNG2  EJA - Ensino Fundamental - Grupo 2    Numérico  Contínuo   
5   EJAFUNG3  EJA - Ensino Fundamental - Grupo 3    Numérico  Contínuo   
6   EJAFUNG4  EJA - Ensino Fundamental - Grupo 4    Numérico  Contínuo   
7   EJAFUNG5  EJA - Ensino Fundamental - Grupo 5    Numérico  Contínuo   
8   EJAMEDG1        EJA - Ensino Médio - Grupo 1    Numérico  Contínuo   
9   EJAMEDG2        EJA - Ensino Médio - Grupo 2    Numérico  Contínuo   
10  EJAMEDG3        EJA - Ensino Médio - Grupo 3    Numérico  Contínuo   
11  EJAMEDG4        EJA - Ensino Médio - Grupo 4    Numérico  Contínuo   
12  EJAMEDG5        EJA - Ensino Médio - Grupo 5    Numérico  Contínuo   

                                      Obs  
0                        Região do Brasil  
1                     Zona Rural e Urbana  
2   Municipal, Estudal, Federal e Privado  
3                                          
4                                          
5                                          
6                                          
7                                          
8                                          
9                                          
10                                         
11                                         
12

In [43]:
# Renomeando as colunas 

colunas = ["GEO", "LOC", "DA", "EJAFUNG1", "EJAFUNG2", "EJAFUNG3", "EJAFUNG4", "EJAFUNG5", "EJAMED1", "EJAMED2", "EJAMED3", "EJAMED4", "EJAMED5"]
afd_eja.columns = colunas

afd_eja


GEO     LOC       DA EJAFUNG1 EJAFUNG2 EJAFUNG3 EJAFUNG4 EJAFUNG5  \
0    Brasil   Total    Total       27      0.5     44.6      2.4     25.5   
1    Brasil  Urbana    Total     35.5      0.7     47.9      2.7     13.2   
2    Brasil   Rural    Total       10      0.1     38.1      1.8       50   
3    Brasil   Total  Federal     68.9        6     17.5      5.5      2.1   
4    Brasil  Urbana  Federal     68.9        6     17.5      5.5      2.1   
..      ...     ...      ...      ...      ...      ...      ...      ...   
590     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
591     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
592     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
593     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
594     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   

    EJAMED1 EJAMED2 EJAMED3 EJAMED4 EJAMED5  
0      56.1     1.1    36.1     3.1     3.6  
1      57.9     1.2    35.1     3.1     2.7  
2      38.4     0.6      46     3.6    11.4  
3      83.3     6.7     6.4     2.9     0.7  
4      83.2     6.8     6.3       3     0.7  
..      ...     ...     ...     ...     ...  
590     NaN     NaN     NaN     NaN     NaN  
591     NaN     NaN     NaN     NaN     NaN  
592     NaN     NaN     NaN     NaN     NaN  
593     NaN     NaN     NaN     NaN     NaN  
594     NaN     NaN     NaN     NaN     NaN  

[595 rows x 13 columns]

In [44]:
afd_eja.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   GEO       588 non-null    object
 1   LOC       588 non-null    object
 2   DA        588 non-null    object
 3   EJAFUNG1  588 non-null    object
 4   EJAFUNG2  588 non-null    object
 5   EJAFUNG3  588 non-null    object
 6   EJAFUNG4  588 non-null    object
 7   EJAFUNG5  588 non-null    object
 8   EJAMED1   588 non-null    object
 9   EJAMED2   588 non-null    object
 10  EJAMED3   588 non-null    object
 11  EJAMED4   588 non-null    object
 12  EJAMED5   588 non-null    object
dtypes: object(13)
memory usage: 60.6+ KB


É possível perceber os valores ausentes nessas 13 colunas seguem sendo identificadas como object. Portanto, para resolver esse problema irei substituir esses valores ausentes resolvi optar pela técnica de imputação Mediana, pela sua robustêz e por ser pouco afetada pelos outliers que podem sugir se eu usasse a média. 

In [45]:
# Iniciando a substituição de "--" para NaN nas colunas com dados numéricos

col_num = ['EJAFUNG1', 'EJAFUNG2', 'EJAFUNG3', 'EJAFUNG4', 'EJAFUNG5', 'EJAMED1', 'EJAMED2', 'EJAMED3', 'EJAMED4', 'EJAMED5'] #minhas colunas numéricas

In [65]:
afd_eja = afd_eja.copy()

In [68]:
#Substituindo "--" por nan apenas nessas colunas

afd_eja.loc[:, col_num] = afd_eja[col_num].replace('--', np.nan)

In [47]:
afd_eja

GEO     LOC       DA EJAFUNG1 EJAFUNG2 EJAFUNG3 EJAFUNG4 EJAFUNG5  \
0    Brasil   Total    Total     27.0      0.5     44.6      2.4     25.5   
1    Brasil  Urbana    Total     35.5      0.7     47.9      2.7     13.2   
2    Brasil   Rural    Total     10.0      0.1     38.1      1.8     50.0   
3    Brasil   Total  Federal     68.9      6.0     17.5      5.5      2.1   
4    Brasil  Urbana  Federal     68.9      6.0     17.5      5.5      2.1   
..      ...     ...      ...      ...      ...      ...      ...      ...   
590     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
591     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
592     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
593     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   
594     NaN     NaN      NaN      NaN      NaN      NaN      NaN      NaN   

    EJAMED1 EJAMED2 EJAMED3 EJAMED4 EJAMED5  
0      56.1     1.1    36.1     3.1     3.6  
1      57.9     1.2    35.1     3.1     2.7  
2      38.4     0.6    46.0     3.6    11.4  
3      83.3     6.7     6.4     2.9     0.7  
4      83.2     6.8     6.3     3.0     0.7  
..      ...     ...     ...     ...     ...  
590     NaN     NaN     NaN     NaN     NaN  
591     NaN     NaN     NaN     NaN     NaN  
592     NaN     NaN     NaN     NaN     NaN  
593     NaN     NaN     NaN     NaN     NaN  
594     NaN     NaN     NaN     NaN     NaN  

[595 rows x 13 columns]

In [71]:
afd_eja.info() # as colunas já não estão mais todas tipo object

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   GEO       588 non-null    object 
 1   LOC       588 non-null    object 
 2   DA        588 non-null    object 
 3   EJAFUNG1  595 non-null    float64
 4   EJAFUNG2  595 non-null    float64
 5   EJAFUNG3  595 non-null    float64
 6   EJAFUNG4  595 non-null    float64
 7   EJAFUNG5  595 non-null    float64
 8   EJAMED1   595 non-null    float64
 9   EJAMED2   595 non-null    float64
 10  EJAMED3   595 non-null    float64
 11  EJAMED4   595 non-null    float64
 12  EJAMED5   595 non-null    float64
dtypes: float64(10), object(3)
memory usage: 60.6+ KB


In [72]:
# Verificando quantos NaN consta nas colunas 

dados_nan = afd_eja [["GEO", "LOC", "DA", "EJAFUNG1", "EJAFUNG2", "EJAFUNG3", "EJAFUNG4", "EJAFUNG5", "EJAMED1", "EJAMED2", "EJAMED3", "EJAMED4", "EJAMED5"]].isnull().sum()

print(dados_nan)

GEO         7
LOC         7
DA          7
EJAFUNG1    0
EJAFUNG2    0
EJAFUNG3    0
EJAFUNG4    0
EJAFUNG5    0
EJAMED1     0
EJAMED2     0
EJAMED3     0
EJAMED4     0
EJAMED5     0
dtype: int64


In [73]:
# Variáveis Numéricas

round(afd_eja.describe(), 2)

EJAFUNG1  EJAFUNG2  EJAFUNG3  EJAFUNG4  EJAFUNG5  EJAMED1  EJAMED2  \
count    595.00    595.00    595.00     595.0    595.00   595.00   595.00   
mean      34.23      0.64     47.90       2.6     13.24    55.33     1.82   
std       18.23      0.94     13.99       4.7     14.78    18.89     3.01   
min        0.00      0.00      0.00       0.0      0.00     0.00     0.00   
25%       21.20      0.20     42.40       1.5      3.40    43.30     0.40   
50%       34.00      0.40     47.90       2.1      7.80    54.45     0.80   
75%       42.90      0.80     55.25       2.6     16.95    69.35     1.70   
max      100.00      7.60    100.00     100.0     69.60   100.00    20.00   

       EJAMED3  EJAMED4  EJAMED5  
count   595.00   595.00   595.00  
mean     34.57     3.26     4.40  
std      17.08     6.26     6.84  
min       0.00     0.00     0.00  
25%      25.40     1.40     0.60  
50%      36.00     2.30     2.20  
75%      45.55     3.40     5.10  
max      80.00   100.00    65.20

In [74]:
# Variáveis Categóricas 

afd_eja.describe(include = [object])

GEO    LOC     DA
count      588    588    588
unique      33      3      6
top     Brasil  Total  Total
freq        18    197     99

In [75]:
afd_eja[["GEO", "LOC", "DA", "EJAFUNG1", "EJAFUNG2", "EJAFUNG3", "EJAFUNG4", "EJAFUNG5", "EJAMED1", "EJAMED2", "EJAMED3", "EJAMED4", "EJAMED5"]].describe().round(2)

EJAFUNG1  EJAFUNG2  EJAFUNG3  EJAFUNG4  EJAFUNG5  EJAMED1  EJAMED2  \
count    595.00    595.00    595.00     595.0    595.00   595.00   595.00   
mean      34.23      0.64     47.90       2.6     13.24    55.33     1.82   
std       18.23      0.94     13.99       4.7     14.78    18.89     3.01   
min        0.00      0.00      0.00       0.0      0.00     0.00     0.00   
25%       21.20      0.20     42.40       1.5      3.40    43.30     0.40   
50%       34.00      0.40     47.90       2.1      7.80    54.45     0.80   
75%       42.90      0.80     55.25       2.6     16.95    69.35     1.70   
max      100.00      7.60    100.00     100.0     69.60   100.00    20.00   

       EJAMED3  EJAMED4  EJAMED5  
count   595.00   595.00   595.00  
mean     34.57     3.26     4.40  
std      17.08     6.26     6.84  
min       0.00     0.00     0.00  
25%      25.40     1.40     0.60  
50%      36.00     2.30     2.20  
75%      45.55     3.40     5.10  
max      80.00   100.00    65.20

In [76]:
# Substituição dos nan pela mediana das colunas
afd_eja.loc[:, col_num] = afd_eja[col_num].apply(lambda x: x.fillna(x.median()))

**Verificando se o replace dos nan pela mediana das colunas funcionou**

In [77]:
afd_eja[col_num].isnull().sum()

EJAFUNG1    0
EJAFUNG2    0
EJAFUNG3    0
EJAFUNG4    0
EJAFUNG5    0
EJAMED1     0
EJAMED2     0
EJAMED3     0
EJAMED4     0
EJAMED5     0
dtype: int64

In [78]:
afd_eja.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 595 entries, 0 to 594
Data columns (total 13 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   GEO       588 non-null    object 
 1   LOC       588 non-null    object 
 2   DA        588 non-null    object 
 3   EJAFUNG1  595 non-null    float64
 4   EJAFUNG2  595 non-null    float64
 5   EJAFUNG3  595 non-null    float64
 6   EJAFUNG4  595 non-null    float64
 7   EJAFUNG5  595 non-null    float64
 8   EJAMED1   595 non-null    float64
 9   EJAMED2   595 non-null    float64
 10  EJAMED3   595 non-null    float64
 11  EJAMED4   595 non-null    float64
 12  EJAMED5   595 non-null    float64
dtypes: float64(10), object(3)
memory usage: 60.6+ KB


In [79]:
afd_eja

GEO     LOC       DA  EJAFUNG1  EJAFUNG2  EJAFUNG3  EJAFUNG4  \
0    Brasil   Total    Total      27.0       0.5      44.6       2.4   
1    Brasil  Urbana    Total      35.5       0.7      47.9       2.7   
2    Brasil   Rural    Total      10.0       0.1      38.1       1.8   
3    Brasil   Total  Federal      68.9       6.0      17.5       5.5   
4    Brasil  Urbana  Federal      68.9       6.0      17.5       5.5   
..      ...     ...      ...       ...       ...       ...       ...   
590     NaN     NaN      NaN      34.0       0.4      47.9       2.1   
591     NaN     NaN      NaN      34.0       0.4      47.9       2.1   
592     NaN     NaN      NaN      34.0       0.4      47.9       2.1   
593     NaN     NaN      NaN      34.0       0.4      47.9       2.1   
594     NaN     NaN      NaN      34.0       0.4      47.9       2.1   

     EJAFUNG5  EJAMED1  EJAMED2  EJAMED3  EJAMED4  EJAMED5  
0        25.5    56.10      1.1     36.1      3.1      3.6  
1        13.2    57.90      1.2     35.1      3.1      2.7  
2        50.0    38.40      0.6     46.0      3.6     11.4  
3         2.1    83.30      6.7      6.4      2.9      0.7  
4         2.1    83.20      6.8      6.3      3.0      0.7  
..        ...      ...      ...      ...      ...      ...  
590       7.8    54.45      0.8     36.0      2.3      2.2  
591       7.8    54.45      0.8     36.0      2.3      2.2  
592       7.8    54.45      0.8     36.0      2.3      2.2  
593       7.8    54.45      0.8     36.0      2.3      2.2  
594       7.8    54.45      0.8     36.0      2.3      2.2  

[595 rows x 13 columns]

Exportando o DF afd_eja

In [80]:
afd_eja.to_csv("afd_eja.csv", sep=';', decimal=',', index=False)

In [81]:
# Fazendo uma cópia do meu df original

df = afd_eja.copy()

In [82]:
afd_eja.groupby("LOC")[["EJAFUNG1", "EJAMED1"]].median().round(2)

EJAFUNG1  EJAMED1
LOC                      
Rural       34.0    54.45
Total       34.0    56.70
Urbana      35.4    58.90

In [83]:
afd_eja.groupby("GEO")[["EJAFUNG1", "EJAMED1"]].median().round(2)

EJAFUNG1  EJAMED1
GEO                                   
Acre                    13.40    37.70
Alagoas                 19.50    46.80
Amapá                   39.20    65.10
Amazonas                18.65    50.85
Bahia                   21.30    46.15
Brasil                  30.25    51.15
Ceará                   16.75    46.75
Centro-Oeste            34.00    52.30
Distrito Federal        40.50    76.30
Espírito Santo          53.55    65.70
Goiás                   31.55    43.00
Maranhão                12.65    40.80
Mato Grosso             24.95    50.40
Mato Grosso do Sul      41.00    60.00
Minas Gerais            41.10    54.45
Nordeste                21.85    50.15
Norte                   30.40    45.60
Paraná                  34.00    69.95
Paraíba                 29.90    65.60
Pará                    38.40    69.65
Pernambuco              24.90    44.75
Piauí                   25.05    57.50
Rio Grande do Norte     33.30    64.40
Rio Grande do Sul       43.45    59.25
Rio de Janeiro          54.75    74.60
Rondônia                34.00    54.35
Roraima                 35.05    54.45
Santa Catarina          62.55    64.10
Sergipe                 34.00    70.70
Sudeste                 47.25    54.45
Sul                     43.70    63.45
São Paulo               39.85    52.55
Tocantins               15.15    52.50

In [84]:
afd_eja.groupby(["GEO","DA"])[["EJAFUNG1", "EJAMED1"]].median().round(2)

EJAFUNG1  EJAMED1
GEO       DA                          
Acre      Estadual       24.2    31.10
          Federal        34.0    54.45
          Municipal       2.8    54.45
          Privada         7.2    38.70
          Pública        13.5    31.10
...                       ...      ...
Tocantins Federal        34.0    73.30
          Municipal      11.2    25.00
          Privada        10.1    52.20
          Pública        16.5    52.50
          Total          13.5    52.50

[197 rows x 2 columns]

In [85]:
afd_eja.groupby(["GEO", "LOC", "DA"])[["EJAFUNG1", "EJAMED1"]].median().round(2)

EJAFUNG1  EJAMED1
GEO       LOC    DA                          
Acre      Rural  Estadual        9.5    12.50
                 Federal        34.0    54.45
                 Municipal       0.9    54.45
                 Pública         3.9    12.50
                 Total           3.9    12.50
...                              ...      ...
Tocantins Urbana Federal        34.0    73.30
                 Municipal      15.4    25.00
                 Privada        10.0    52.20
                 Pública        21.1    58.90
                 Total          14.9    58.50

[588 rows x 2 columns]

### Visualizando os dados - EJA

Dicionário do dataframe afd_eja (adequação da formação docente - EJA)

- GEO: Unidade geográfica
- LOC: Zona rural ou urbana
- DA: Dependência administrativa (estadual, municipal, federal e privado)
- EJAFUNG1 a EJAFUNG5: Educação de Jovens e Adultos no Ensino Fundamental nos grupos 1 a 5
- EJAMED1 a EJAMED5: Educação de Jovens e Adultos no Ensino Médio nos grupos de 1 a 5
- Grupo 1: Representa docentes que possuem a formação adequada
- Grupo 2 a Grupo 5: Representa docentes sem formação adequada em diferentes níveis

*1. Qual o percentual de docentes com formação adequada na EJA do Ensino Fundamental e Médio por zona(LOC)*

*2. Analisando separadamente, qual zona apresenta maior discrepância na formação docente para a EJA do Ensino Fundamental? E para a EJA do Ensino Médio?*

*3. Em quais dependências admnistritativas há uma maior incidênica dessa discrepância em relação a EJA do Ensino Fundamental? E no Ensino Médio?*

*4. Há alguma relação entre a unidade geográfica e a localização em realação a EJA do Ensino Fundamental? E no Ensino Médio?*

*5. Existe alguma correlação entre a unidade geográfica e a dependencia administrativa em relação à formação docente da EJA no Ensino Fundamental? E no Ensino Médio?*

*6. Ao fazer uma comparação entre o Ensino Fundamental EJA e Ensino Médio EJA, qual podemos observar que há uma maior dispariadade?*

*7. Qual região do Brasil apresenta a maior variação na formação dos docentes da EJA em termos de adequação (Grupos 1 a 5)?*

*8. Qual grupo de formação docente (1, 2, 3, 4 e 5) concentra um maior percentual desses docentes, considerando a zona geográfica?*

*9. Qual grupo de formação docente (Grupo 1, 2, 3, 4, 5) apresenta maior percentual de docentes na EJA, considerando a dependência administrativa?*

*10. Comparando as regiões do Brasil, qual apresenta maior desigualdade na adequação da formação docente para a EJA?*

### Conclusão